packages/@react-spectrum/tag/src/TagGroup.tsx

```diff
@@ -167,6 +172,7 @@ function TagGroup<T extends object>(props: SpectrumTagGroupProps<T>, ref: DOMRef
           )
         }>
         <div
+          style={maxRows != null && tagState.showCollapseButton && isCollapsed ? {maxHeight: tagState.maxHeight, overflow: 'hidden'} : undefined}
```

Comment:
I think the flickering is when it transitions from not having an collapse button to having one for the first time, because there's no maxHeight defined when no collapse button.